bin/utest/pgrp.c

```diff
@@ -45,6 +39,98 @@ static void sa_handler(int signo) {
   sig_delivered = 1;
 }
 
+int test_setpgid_leader(void) {
+  signal(SIGUSR1, sa_handler);
+
+  pid_t cpid = fork();
+  if (cpid == 0) {
+    /* Become session leader. */
+    assert(setsid() == getpid());
+    /* Can't change pgrp of session leader. */
+    assert(setpgid(0, 0));
+    while (!sig_delivered)
+      sched_yield();
+    return 0;
+  }
+
+  /* Wait until child becomes session leader. */
+  while (getsid(cpid) != cpid)
+    sched_yield();
+
+  /* Can't change pgrp of session leader. */
+  assert(setpgid(cpid, getpgid(0)));
+
+  kill(cpid, SIGUSR1);
+
+  int status;
```

Comment:
This code repeats over and over... could we replace it with a function, for instance `assert(child_exitcode(pid, 0))`?